src/saturn_engine/utils/inspect.py

```diff
@@ -149,15 +151,10 @@ def signature(func: t.Callable) -> inspect.Signature:
 
 
 def unwrap_optional(typ: t.Any) -> t.Optional[t.Type]:
-    origin = t.get_origin(typ)
-    if origin is t.Union:
-        target_args = [
-            arg
-            for arg in t.get_args(typ)
-            if not (not t.get_args(arg) and isinstance(None, arg))
-        ]
-        if len(target_args) == 1:
-            return target_args[0]
+    if typing_inspect.is_optional_type(typ):
+        args = typing_inspect.get_args(typ)
```

Comment:
nit but this could have been a 1 liner
return typing_inspect.get_args(typ)[0] if typing_inspect.get_args(typ)

Reply:
you are missing the `else None`, but I would argue that 

```python
args = typing_inspect.get_args(typ)
if args:
  return args[0]
```
is more readable than
```python
return typing_inspect.get_args(typ)[0] if typing_inspect.get_args(typ) else None
``` 

(And it's busting the max line lenght so it's going to get split into three line by the formatter :P )

If you really want to code gold, `return next(typing_inspect.get_args(typ), None)` would have done it, but people are less familiar with this, so KISS it is!